internal/plugin/connectors/http/generic/config_test.go

```diff
@@ -20,16 +21,21 @@ func sampleConfig() []byte {
 `)
 }
 
+func getSampleConfig(t *testing.T) (*config, error) {
+	cfgYAML, err := NewConfigYAML(sampleConfig())
+	if err != nil {
+		assert.Fail(t, "sampleConfig should never fail")
+		return nil, err
+	}
+	cfg, err := newConfig(cfgYAML)
+
+	assert.NoError(t, err)
+	return cfg, err
+}
+
```

Comment:
This is a relatively small instance of it, ofc, but i think there are a couple important conceptual issues here worth addressing.

1. We're mixing concerns.  Currently, this is both a utility for constructing a `config` as well as something that runs some tests.  It wants to be only the former.  We want all our tests to be directly inside `TestXXX` methods.  Fixing this will also let us to remove the paramter `t`.
2. Before this change, we had `sampleConfig`, which was a literal yaml string.  That name was ok because there was nothing to confuse it with.  Since we now have two separate helper functions for closely related but different things -- the literal yaml config, and the `config` _type_ -- we want to give them unambiguous names.  I'd suggest `sampleConfigYAML` and `sampleConfig`.
3. A broader principle related to point 2 -- we never use `getXXX` for value types.   Here's a [short video snippet on this best practice](https://youtu.be/MzTcsI6tn-0?t=1457).  Instead, the name should always be a noun that describes the thing being returned.

Reply:
Great suggestions! Thanks for the explanations and pointers.